src/main/java/de/tum/in/www1/artemis/web/rest/BuildPlanResource.java

```diff
@@ -51,4 +65,50 @@ public ResponseEntity<String> getBuildPlan(@PathVariable Long exerciseId, @Reque
 
         return ResponseEntity.ok().body(buildPlan.getBuildPlan());
     }
+
+    /**
+     * Returns the build plan for a given programming exercise.
+     *
+     * @param exerciseId the exercise for which the build plan should be retrieved
+     * @return the build plan stored in the database
+     */
+    @GetMapping(BUILD_PLAN + "/for-editor")
+    @PreAuthorize("permitAll()")
```

Comment:
Since only editors should access the build plan, there should already be a first access control check here.
```suggestion
    @PreAuthorize("hasRole('EDITOR')")
```

Reply:
Thanks for looking at the code! Good call, this was an oversight from when I copy-pasted the other `getBuildPlan` method.